modules/standard/IO.chpl

```diff
@@ -3090,7 +3090,13 @@ private inline proc _read_one_internal(_channel_internal:qio_channel_ptr_t,
   // to stop reading if there was an error.
   qio_channel_clear_error(_channel_internal);
 
-  x.readThis(reader);
+  try {
+    x.readThis(reader);
+  } catch err {
+    //
+    // TODO: What to do with the caught error? Propagate back up?
+    //
```

Comment:
Please set the error on the channel using qio_channel_set_error etc to *something* for now

Reply:
Just to double check, this is using `_qio_channel_set_error_unlocked`?

Reply:
Ah, has to be, comment mentions we already have the lock (I can't read).